applications/asset_tracker_v2/src/modules/modem_module.c

```diff
@@ -293,39 +293,33 @@ static int static_modem_data_get(void)
 
 	check_modem_fw_version();
 
-	struct modem_module_event *modem_module_event =
-			new_modem_module_event();
-
-	modem_module_event->data.modem_static.app_version =
-			CONFIG_ASSET_TRACKER_V2_APP_VERSION;
-
-	modem_module_event->data.modem_static.board_version =
-			modem_param.device.board;
+	struct modem_module_event *modem_module_event = new_modem_module_event();
 
-	modem_module_event->data.modem_static.modem_fw =
-			modem_param.device.modem_fw.value_string;
+	modem_module_event->data.modem_static.nw_mode_ltem = modem_param.network.lte_mode.value;
+	modem_module_event->data.modem_static.nw_mode_nbiot = modem_param.network.nbiot_mode.value;
+	modem_module_event->data.modem_static.nw_mode_gps = modem_param.network.gps_mode.value;
+	modem_module_event->data.modem_static.band = modem_param.network.current_band.value;
 
-	modem_module_event->data.modem_static.iccid =
-			modem_param.sim.iccid.value_string;
+	strncpy(modem_module_event->data.modem_static.app_version,
+		CONFIG_ASSET_TRACKER_V2_APP_VERSION,
+		sizeof(modem_module_event->data.modem_static.app_version) - 1);
```

Comment:
Note that if the destination buffer is too small to hold the string in the source buffer, `strncpy` will fill the buffer and not null terminate it. In this case, `modem_module_event->data.modem_static.app_version` holds random data from earlier, so we risk passing on an non-terminated string. Either something like `modem_module_event->data.modem_static.app_version[strlen(CONFIG_ASSET_TRACKER_V2_APP_VERSION)] = '\0'` or `memset(modem_module_event->data.modem_static.app_version, '\0', ARRAY_SIZE(modem_module_event->data.modem_static.app_version))` can be used to protect against that

Reply:
Good point. Definitely missed that detail. Ill explicitly add null termination at the end of the destination buffer. I think ill use ```sizeof``` instead of ```strlen``` if in case, the buffer is not null terminated after the ```strncpy```.